File: joao pedro/Problema0289/Problema0289.py
numero=[]
lugares=[]
vago=[]
pedidos=[]
x=0
y=0

n=int(input("Digite a quantidade de voos: "))

for i in range (n):
    print("Digite o numero do voo ",i,": ")
    numeros=int(input())
    numero.insert(i,numeros)
    print("Digite a quantidade de lugares disponiveis no voo",i,": ")
    lugar=int(input())
    lugares.insert(i,lugar)
    if lugares[i]<=0:
        print("Voo lotado")
        
    for j in range (lugares[i]):
        print("Deseja realizar uma reserva para o voo ",i," ?([1]Sim/[2]Nao/[-1]Para encerrar]): ")
        x=int(input())
        if x==1:
            if y==0:
                pedidos.insert(i,0)
                
                y=1
            if y==1:
                pedidos[i]+=1
            
            print("** Pedido efetuado **")
        if x==-1:
            print("** Encerrando programa **")
            exit()
    y=0
    
for k in range (n):
    vago.insert(k,0)
    if lugares[k]!=0:
        
        vago[k]=lugares[k]-pedidos[k]
        print("Numero do voo: ",numero[k])
        print("Passageiros: ",pedidos[k])
        print("Lugares vagos: ",vago[k])
    if lugares[k]==0:
        print("Numero do voo: ",numero[k])
        print("Passageiros: 0")
        print("Lugares vagos: 0")

File: joao pedro/Problema0289/Problema0289.js
const prompt = require("prompt-sync")({ sigint: true });

var x=0, numero=[],lugares=[],vago=[],pedidos=[],n,y=0;

n=Number(prompt("Digite a quantidade de voos: "))

for(i=0;i<n;i++){
    numero[i]=Number(prompt("Digite o numero do voo "+i+": "));
    
    lugares[i]=Number(prompt("Digite a quantidade de lugares disponiveis no voo "+i+": "));
    
    if(lugares[i]<=0){
        console.log("Voo lotado");
        pedidos[i]=0;
    }
    for(j=0;j<lugares[i];j++){
        x=Number(prompt("Deseja realizar uma reserva para o voo "+i+" ?([1]Sim/[2]Nao/[-1]Para encerrar]): "));
        
        if(x==1){
          if(y==0){
            pedidos[i]=0;
            
            y=1;}
            if(y==1){
                pedidos[i]++;
                console.log("** Pedido efetuado **");
            }
        }
        if(x==-1){
            console.log("** Encerrando programa **");
            return;
        }
        y=0;
    }
    

}
for(i=0;i<n;i++){
    
    vago[i]=lugares[i]-pedidos[i];
    console.log("Numero do voo: "+numero[i]+". Passageiros: "+pedidos[i]+". Lugares vagos: "+vago[i]);
}